src/VirtualList/VirtualList.js

```diff
@@ -27,15 +29,18 @@ export default class VirtualList extends Util.mixin(BindMixin) {
   }
 
   componentWillReceiveProps(nextProps) {
+    if (this.props.container !== nextProps.container) {
+      this.props.container.removeEventListener('scroll', this.onScrollDebounced);
+      nextProps.container.addEventListener('scroll', this.onScrollDebounced);
+    }
     let state = this.getVirtualState(nextProps);
     this.setState(state);
   }
 
   componentWillMount() {
-    this.onScrollDebounced = Util.debounce(
+    this.onScrollDebounced = Util.throttle(
       this.onScroll,
-      this.props.scrollDelay,
-      false
+      this.props.scrollDelay
```

Comment:
The comment for explaining `scrollDelay` still mentions `debounce`, let's update that too